sdk/keyvault/Azure.Security.KeyVault.Administration/src/KeyVaultPreBackupOperation.cs

```diff
@@ -0,0 +1,177 @@
+
+        /// <summary>
+        /// Creates an instance of a KeyVaultPreBackupOperation from a previously started operation. <see cref="UpdateStatus(CancellationToken)"/>, <see cref="UpdateStatusAsync(CancellationToken)"/>,
+        ///  <see cref="WaitForCompletionAsync(CancellationToken)"/>, or <see cref="WaitForCompletionAsync(TimeSpan, CancellationToken)"/> must be called
+        /// to re-populate the details of this operation.
+        /// </summary>
+        /// <param name="client">An instance of <see cref="KeyVaultBackupClient" />.</param>
+        /// <param name="id">The <see cref="Id" /> from a previous <see cref="KeyVaultPreBackupOperation" />.</param>
+        /// <exception cref="ArgumentNullException"><paramref name="id"/> or <paramref name="client"/> is null.</exception>
+        public KeyVaultPreBackupOperation(KeyVaultBackupClient client, string id)
+        {
+            Argument.AssertNotNull(id, nameof(id));
+            Argument.AssertNotNull(client, nameof(client));
+
+            _client = client;
+            _id = id;
+        }
+
+        /// <summary>
+        /// Initializes a new instance of a KeyVaultPreBackupOperation.
+        /// </summary>
+        /// <param name="client">An instance of <see cref="KeyVaultBackupClient" />.</param>
+        /// <param name="response">The <see cref="ResponseWithHeaders{T,THeaders}"/> returned from <see cref="KeyVaultBackupClient.StartPreBackup(Uri, string, CancellationToken)"/> or <see cref="KeyVaultBackupClient.StartPreBackupAsync(Uri, string, CancellationToken)"/>.</param>
+        /// <exception cref="InvalidOperationException"> The server operation does not contains an Id</exception>
+        internal KeyVaultPreBackupOperation(KeyVaultBackupClient client, ResponseWithHeaders<AzureSecurityKeyVaultAdministrationPreFullBackupHeaders> response)
+        {
+            _client = client;
+            _response = response;
+            _retryAfterSeconds = response.Headers.RetryAfter;
+            _id = response.Headers.JobId() ?? throw new InvalidOperationException("The response does not contain an Id");
+        }
+
+        /// <summary>
+        /// Initializes a new instance of a KeyVaultPreBackupOperation for mocking purposes.
+        /// </summary>
+        /// <param name="value">The <see cref="FullBackupDetailsInternal" /> that will be returned from <see cref="Value" />.</param>
+        /// <param name="response">The <see cref="Response" /> that will be returned from <see cref="GetRawResponse" />.</param>
+        /// <param name="client">An instance of <see cref="KeyVaultBackupClient" />.</param>
+        internal KeyVaultPreBackupOperation(FullBackupDetailsInternal value, Response response, KeyVaultBackupClient client)
+        {
+            Argument.AssertNotNull(value, nameof(value));
+            Argument.AssertNotNull(response, nameof(response));
+            Argument.AssertNotNull(client, nameof(client));
```

Comment:
Should document all known exceptions. You can pretty much copy and paste what all I've typed over the years and just change the `paramref`s.